lib/instrument/transformer/nskeyed.js

```diff
@@ -202,12 +210,108 @@ class XCTCapabilities extends BaseArchiveHandler {
   }
 }
 
+/**
+ * Some of the plist classes may not be included and should not be cared.
+ * Using this as a default decoder for unknown entities to avoid interrupting decode process.
+ */
+class NSIgnore extends BaseArchiveHandler {
+  decodeArchive() {
+    return null;
+  }
+}
+
 class NSNull extends BaseArchiveHandler {
+  constructor() {
+    super();
+    this.additionClasses = [NS_OBJECT];
+  }
   decodeArchive () {
     return null;
   }
 }
 
+class XCTestConfiguration extends BaseArchiveHandler {
+  static _default = {
+    aggregateStatisticsBeforeCrash: {
+      XCSuiteRecordsKey: {}
+    },
+    automationFrameworkPath: '/Developer/Library/PrivateFrameworks/XCTAutomationSupport.framework',
+    baselineFileRelativePath: undefined,
+    baselineFileURL: undefined,
+    defaultTestExecutionTimeAllowance: undefined,
+    disablePerformanceMetrics: false,
+    emitOSLogs: false,
+    formatVersion: 2,
+    gatherLocalizableStringsData: false,
+    initializeForUITesting: true,
+    maximumTestExecutionTimeAllowance: undefined,
+    productModuleName: 'WebDriverAgentRunner',
+    randomExecutionOrderingSeed: undefined,
+    reportActivities: true,
+    reportResultsToIDE: true,
+    systemAttachmentLifetime: 2,
+    targetApplicationArguments: [],
+    targetApplicationBundleID: undefined,
+    targetApplicationEnvironment: undefined,
+    targetApplicationPath: '/KEEP-THIS-NOT-EMPTY/KEEP-THIS-NOT-EMPTY',
+    testApplicationDependencies: {},
+    testApplicationUserOverrides: undefined,
+    testBundleRelativePath: undefined,
+    testExecutionOrdering: 0,
+    testTimeoutsEnabled: false,
+    testsDrivenByIDE: false,
+    testsMustRunOnMainThread: true,
+    testsToRun: undefined,
+    testsToSkip: undefined,
+    treatMissingBaselinesAsFailures: false,
+    userAttachmentLifetime: 1
+  }
+  /**
+   *
+   * @param {object} data
```

Comment:
Please use typedef in order to describe complex arguments. For example, check https://github.com/appium/appium-ios-device/blob/276f3fadb2c75dfa9caa42ed2214b2a48c4e5848/lib/mcinstall/index.js#L21